test/common/http/conn_pool_grid_test.cc

```diff
@@ -126,12 +127,66 @@ TEST_F(ConnectivityGridTest, FailureThenSuccess) {
   // onPoolReady should be passed from the pool back to the original caller.
   ASSERT_NE(grid_.callbacks(), nullptr);
   EXPECT_CALL(callbacks_.pool_ready_, ready());
-  grid_.callbacks()->onPoolReady(encoder_, host_, info_, absl::nullopt);
+  grid_.callbacks(1)->onPoolReady(encoder_, host_, info_, absl::nullopt);
+}
+
+// Test both connections happening in parallel and the second connecting.
+TEST_F(ConnectivityGridTest, FailureThenSuccessParallelSecondConnects) {
```

Comment:
I think this test should be called TimeoutThenSuccessParallelSecondConnects or something since the first one didn't fail. Some point for other tests in this file.